crates/oci-distribution/src/lib.rs

```diff
@@ -145,6 +153,35 @@ impl Client {
             )),
         }
     }
+
+    /// Pull a single layer from an OCI registy.
+    ///
+    /// This pulls the layer for a particular image that is identified by
+    /// the given digest. The image reference is used to find the
+    /// repository and the registry, but it is not used to verify that
+    /// the digest is a layer inside of the image. (The manifest is
+    /// used for that.)
+    pub async fn pull_layer<T: std::io::Write>(
```

Comment:
@rylev So here's where I am a little stuck: I want to be able to write simple unit tests that consume the stream into a buffer so I can test the size and such. But in most real-world cases, the `Write` implementation should be async. I can't figure out how to satisfy the first using `tokio::io::AsyncWrite`. But it does look like `tokio::fs::File` and family do support `Write`. Any suggestions on what I should do? I've also marked the related test with a comment.

Reply:
This should be `AsyncWrite`. What errors do you get when you try with `AsyncWrite`?

Reply:
I just realized that I needed `Write + AsyncWrite`. Then it started working. Thanks!